addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3187,6 +3188,10 @@ var SnippetsMenu = Widget.extend({
      */
     _onSnippetDragAndDropStop: async function (ev) {
         this.snippetEditorDragging = false;
+        const visibleConditionalEls = this.snippetEditors
+            .map(snippetEditor => snippetEditor.$target[0])
+            .filter(targetEl => targetEl.dataset["visibility"] === "conditional" &&
+                                !targetEl.classList.contains("o_conditional_hidden"));
```

Comment:
Also after that you itterate again over `visibleConditionalEls` to remove a class and an attribute. You can also do that in the single loop you will add :+1:

Reply:
> Here you loop two times over `this.snippetEditors`, one to map and one to filter... I think you can do it in one loop like this...
> 
> Or I'm missing something ?

Sure, you are right, I used this `filter` function because on another review, someone told me to use it rather than to use 
```
for() {
     if() {
     }
}
```
but sure, here I also have to use `map()` so it is more efficient to do like you said :+1: 


> Also after that you itterate again over visibleConditionalEls to remove a class and an attribute. You can also do that in the single loop you will add

Not sure to understand your comment here as, if I am not mistaken, I can not do that in the first loop. Indeed the goal this first loop is to store (in `visibleConditionalEls`) the visible elements that have a conditional visibility before the `_destroyEditors()`. Indeed, because of this call, `cleanForSave()` of `snippetEditor` will be called and all the elements with a conditional visibility will be hidden. The goal of the second iteration is to show the elements that were hidden due to the call to `_destroyEditors()` :+1: